lib/health_cards/health_card.rb

```diff
@@ -1,214 +1,58 @@
-      StructureDefinition|StructureMap|Subscription|Substance|SubstanceNucleicAcid|SubstancePolymer|SubstanceProtein|
-      SubstanceReferenceInformation|SubstanceSourceMaterial|SubstanceSpecification|SupplyDelivery|SupplyRequest|Task|
-      TerminologyCapabilities|TestReport|TestScript|ValueSet|VerificationResult|VisionPrescription)/
-      [A-Za-z0-9\-.]{1,64}(/_history/[A-Za-z0-9\-.]{1,64})?}x.freeze
-
-    attr_reader :issuer, :nbf, :bundle
-
-    class << self
-      # Creates a HealthCard from a JWS
-      # @param jws [String] the JWS string
-      # @param public_key [HealthCards::PublicKey] the public key associated with the JWS
-      # @param key [HealthCards::PrivateKey] the private key associated with the JWS
-      # @return [HealthCards::HealthCard]
-      def from_jws(jws, public_key: nil, key: nil)
-        jws = JWS.from_jws(jws, public_key: public_key, key: key)
-        from_payload(jws.payload)
-      end
-
-      # Create a HealthCard from a compressed payload
-      # @param payload [String]
-      # @return [HealthCards::HealthCard]
-      def from_payload(payload)
-        json = decompress_payload(payload)
-        bundle_hash = json.dig('vc', 'credentialSubject', 'fhirBundle')
-
-        raise HealthCards::InvalidCredentialError unless bundle_hash
+    extend Forwardable
 
-        bundle = FHIR::Bundle.new(bundle_hash)
-        new(issuer: json['iss'], bundle: bundle)
-      end
-
-      # Decompress an arbitrary payload, useful for debugging
-      # @param payload [String] compressed payload
-      # @return [Hash] Hash built from JSON contents of payload
-      def decompress_payload(payload)
-        inf = Zlib::Inflate.new(-Zlib::MAX_WBITS).inflate(payload)
-        JSON.parse(inf)
-      end
+    attr_reader :jws
 
-      # Compress an arbitrary payload, useful for debugging
-      # @param payload [Object] Any object that responds to to_s
-      # @return A compressed version of that payload parameter
-      def compress_payload(payload)
-        Zlib::Deflate.new(nil, -Zlib::MAX_WBITS).deflate(payload.to_s, Zlib::FINISH)
-      end
+    def_delegator :@jws, :verify
+    def_delegator :@qr_codes, :code_by_ordinal
+    def_delegators :@payload, :bundle, :issuer
 
-      # Sets/Gets the fhir version that will be passed through to the credential created by an instnace of
-      # this HealthCard (sub)class
-      # @param ver [String] FHIR Version supported by this HealthCard (sub)class. Leaving this param out
-      # will only return the current value
-      # value (used as a getter)
-      # @return [String] Current FHIR version supported
-      def fhir_version(ver = nil)
-        if @fhir_version.nil? && ver.nil?
-          @fhir_version = superclass.fhir_version unless self == HealthCards::HealthCard
-        elsif ver
-          @fhir_version = ver
-        end
-        @fhir_version
-      end
+    def initialize(jws)
+      @jws = jws.is_a?(String) ? JWS.from_jws(jws) : jws
```

Comment:
```suggestion
      @jws = JWS.from_jws(jws)
```

`JWS.from_jws` gracefully handles the JWS type (String, HealthCards::JWS, etc.).